src/lib/VirtualScroller.js

```diff
@@ -372,6 +374,53 @@ class VirtualScroller {
         newListEl.style.height = `${this.totalItems * (this.itemHeight + this.margin) + this.margin}px`;
         return newListEl;
     }
+
+    /**
+     * Scrolls the provided row index into view.
+     * @param {number} rowIndex - the index of the row in the overall list
+     * @return {void}
+     */
+    scrollIntoView(rowIndex) {
+        if (!this.scrollingEl || rowIndex < 0 || rowIndex >= this.totalItems) {
+            return;
+        }
+
+        // See if the list item indexed by `rowIndex` is already present
+        const foundItem = Array.prototype.slice.call(this.listEl.children).find((listItem) => {
+            const { bpVsRowIndex } = listItem.dataset;
+            const parsedRowIndex = parseInt(bpVsRowIndex, 10);
+            return parsedRowIndex === rowIndex;
+        });
+
+        if (foundItem) {
+            // If it is already present and visible, do nothing, but if not visible
+            // then scroll it into view
+            if (!this.isVisible(foundItem)) {
```

Comment:
Is it that bad to call scrollIntoView even the thumbnail is already visible? that would allow you to get rid of isVisible and it could be kind of nice to top align the thumbnail even if it's already visible.

Reply:
it's not necessarily bad, but since cross browser support for the "smooth" `scrollIntoView` is not there, I was trying to avoid the slight jumpiness it causes IMO. I can show you tomorrow to see what you think